tools/pylib/boutdata/data.py

```diff
@@ -243,29 +264,78 @@ def __init__(self, path=".", prefix="BOUT.dmp"):
         
         # Available variables
         self.varNames = []
+        self.dimensions = {}
+        self.evolvingVariableNames = []
+
+        # Private variables
+        if self._caching:
+            from collections import OrderedDict
+            self._datacache = OrderedDict()
+            if self._caching is not True:
+                # Track the size of _datacache and limit it to a maximum of _caching
+                try:
+                    # Check that _caching is a number of some sort
+                    float(self._caching)
+                except ValueError:
+                    raise ValueError("BoutOutputs: Invalid value for caching argument. Caching should be either a number (giving the maximum size of the cache in GB), True for unlimited size or False for no caching.")
+                self._datacachesize = 0
+                self._datacachemaxsize = self._caching*1.e9
         
         with DataFile(file_list[0]) as f:
             # Get variable names
             self.varNames = f.keys()
+            for name in f.keys():
+                dimensions = f.dimensions(name)
+                self.dimensions[name] = dimensions
+                if name != "t_array" and "t" in dimensions:
```

Comment:
Why exclude `t_array`?
The time is also evolved ...

Reply:
because I was using `evolvingVariables()` to plot all the output, and don't want to plot `t_array`. Also, since `t_array` is always present, we don't need to query the data files to find out its name.

The use case I mean would be something like:
```
data = BoutOutputs()
showdata( [ data[x][:,:,0,:] for x in data.evolvingVariables() ] )
```